teps/0121-unified-retries-for-runtime-objects.md

```diff
@@ -0,0 +1,421 @@
+
+It lists several use cases:
+- the momentary loss of network connectivity to components and services
+- the temporary unavailability of a service
+- timeouts that arise when a service is busy
+
+Which all make sense to us as a Cloud Native CI/CD system.
+
+### Requirements
+
+- The behavior of retry should be unified across all runtime objects including TaskRun, CustomRun, potentially as well as PipelineRun once [TEP-0056](https://github.com/tektoncd/community/blob/main/teps/0056-pipelines-in-pipelines.md) is implemented.
+- The implement pattern of the `retry` functionality should be unified across all run time objects.
+
+<!--
+Describe constraints on the solution that must be met, such as:
+- which performance characteristics that must be met?
+- which specific edge cases that must be handled?
+- which user scenarios that will be affected and must be accommodated?
+-->
+
+## Related Work
+
+In the CI/CD industry, most systems support retry, some of them enables users to configure the retry strategy[^retry-strategy].
+
+Typically, a retry strategy needs to specify:
+1. When to retry
+2. The amount of attempts
+3. Actions to take after a failed attempt
+4. Timeout of each attempt
+
+
+|  | [Retry Step in GA](https://github.com/marketplace/actions/retry-step) | [Retry Action in GA](https://github.com/marketplace/actions/retry-action) | [Argo Workflow](https://argoproj.github.io/argo-workflows/retries/#retry-policies) | [GitLab](https://docs.gitlab.com/ee/ci/yaml/#retry) | [Ansible](https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#retrying-a-task-until-a-condition-is-met)| [Concourse](https://concourse-ci.org/attempts-step.html#attempts-step) | Bamboo | Dagger |
+|:---|:---|:---|:---|:---|:---|:---|:---|:---|
+| **When to Retry** | configurable | on failure |configurable|configurable|[always retry, conditional stop](https://github.com/ansible/ansible/pull/76101) [^ansible-conditional-stop]|configurable|-|-|
+| **Attempts amount** |supported|supported|supported|supported|supported|supported|-|-|
+| **Timeout for each attempt** |supported|supported|supported|[supported](https://docs.gitlab.com/ee/ci/yaml/#retrywhen)|supported|supported|-|-|
+| **Timeout for all attempts** |[supported](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes)|[supported](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepstimeout-minutes)|[?](https://stackoverflow.com/questions/71424429/argo-workflowtemplate-timeout)|-|-|-|-|-|
```

Comment:
you can remove bamboo and dagger if there's no related work in those solutions

Reply:
SG, done!